recipes/elfutils/all/conanfile.py

```diff
@@ -33,111 +38,151 @@ class ElfutilsConan(ConanFile):
         "with_sqlite3": False,
     }
 
-    generators = "pkg_config"
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
 
-    _autotools = None
-    _source_subfolder = "source_subfolder"
+    def disable_werror_flags(self):
+        # This function patches `eu.am` so that warnings are not treated as errors
+        # otherwise the library cannot be built due to warnings related to dereferencing pointers
+        path = os.path.join(self.source_folder, "config", "eu.am")
+        find = '$(if $($(*F)_no_Werror),,-Werror)'
+        repl = '$(if $($(*F)_no_Werror),,)'
+        replace_in_file(self, path, find, repl)
+
+    def enable_static_elfutils(self):
+        # This function patches `configure.ac` to allow building elfutils as a static library
+        path = os.path.join(self.source_folder, "configure.ac")
+        find1 = r'AM_CONDITIONAL(INSTALL_ELFH, test "$install_elfh" = yes)'
+        repl1 = f'{find1}\n\n'
+        repl1 += r'AC_MSG_CHECKING([whether to build elfutils as a static library])' "\n"
+        repl1 += r'AC_ARG_ENABLE([static],' "\n"
+        repl1 += r'AS_HELP_STRING([--enable-static],[build elfutils as static libraries]),' "\n"
+        repl1 += r'[static_elfutils=$enableval], [static_elfutils=no])' "\n\n"
+        repl1 += r'static_elfutils=$static_elfutils' "\n\n"
+        repl1 += r'if test "$static_elfutils" = yes ; then' "\n"
+        repl1 += r'  AC_MSG_RESULT([yes])' "\n"
+        repl1 += r'else' "\n"
+        repl1 += r'  if test "$use_gprof" = yes -o "$use_gcov" = yes ; then' "\n"
+        repl1 += r'    AC_MSG_ERROR([cannot build a shared elfutils with gprof or gcov enabled])' "\n"
+        repl1 += r'  fi' "\n"
+        repl1 += r'  AC_MSG_RESULT([no])' "\n"
+        repl1 += 'fi' "\n\n"
+        find2 = r'test "$use_gprof" = yes -o "$use_gcov" = yes])'
+        repl2 = r'test "$static_elfutils" = yes])'
+        replace_in_file(self, path, find1, repl1)
+        replace_in_file(self, path, find2, repl2)
```

Comment:
As @uilianries said above (https://github.com/conan-io/conan-center-index/pull/17272/files#r1245095386), it would be better to keep the patches instead.

Reply:
I'll apply the changes directly in this branch to get it merged soon